### PR TITLE
Add ARIMA to AutoML

### LINKED ISSUE #1946: Time series: add ARIMA to automl
#1498 defined an ARIMA estimator for regression. Next step is to refine the parameters and automl ranges, perf test and add to automl.

### PR DESCRIPTION
**Pull request by ParthivNaresh**
Fixes #1946

### CODE REVIEW

**Comment by ParthivNaresh** [comment]
Performance tests [here](https://alteryx.quip.com/RPH2Ai8l53gO/ARIMA-Regression-Performance-Tests)

File: evalml/pipelines/components/transformers/scalers/standard_scaler.py
Comment (ParthivNaresh) [inline evalml/pipelines/components/transformers/scalers/standard_scaler.py:36]: Necessary otherwise datetime columns are standardized which results in an error.

**Review by jeremyliweishih — APPROVED** [review]
This PR looks good to me - just left a clarifying question. I wonder if there's a way to clean up the tests due to the exception in pipeline generation for ARIMA but I couldn't think of anything off the top of my head. Moving forward with TS problems for performance testing, I'm hoping that we can add datasets with seasonality or other characteristics that models like ARIMA can leverage.

File: evalml/tests/component_tests/test_arima_regressor.py
Comment (ParthivNaresh) [reply evalml/tests/component_tests/test_arima_regressor.py]: And to address your question, I agree entirely. The more time series datasets we can add with seasonality/various trends/etc, the better AutoARIMA can perform. Currently it can hold its own against some of the other regressor models but in the future it should most definitely outperform for most TS datasets.

**Review by bchen1116 — APPROVED** [review]
LGTM! Agree that we should figure out why the arima model fails on some data sets in the perf tests.

**Comment by ParthivNaresh** [comment]
Due to `pmdarima` conda installation issues, #2237 has been filed

File: evalml/pipelines/utils.py
Comment (dsherry) [inline evalml/pipelines/utils.py:88]: Nice

File: evalml/tests/component_tests/test_utils.py
Comment (dsherry) [inline evalml/tests/component_tests/test_utils.py:35]: Nit-pick: flatten
```python
if has_minimal_dependencies:
     n_components = 37
elif is_using_conda:
    n_components = 48
elif is_running_py_39_or_above:
    n_components = 49
else:
    n_components = 47
assert len(all_components()) == n_components
```

File: evalml/tests/conftest.py
Comment (dsherry) [inline evalml/tests/conftest.py:115]: We have a separate conda feedstock PR to add this there, right?

I'm guessing we need that to be merged first in order for CI to work properly, yes?

File: evalml/tests/utils_tests/test_dependencies.py
Comment (dsherry) [inline evalml/tests/utils_tests/test_dependencies.py:21]: Nit-pick: could you merge this in with line 17?
```python
if ((module == 'sktime' and is_running_py_39_or_above) or
    (module == 'pmdarima' and is_using_conda)):
    ...
```

File: requirements.txt
Comment (dsherry) [inline requirements.txt:14]: Now that we've disabled arima for conda installs while pmdarima sorts out the dependency issue: is there any reason we shouldn't do
```python
pmdarima>=1.8.0
```
here?

**Review by dsherry — APPROVED** [review]
LGTM! Let me know when we've got the corresponding conda-forge change approved/merged and are ready to merge this PR and I will gladly do so :)

Other comments blocking merge:
* Please add a note to the ARIMA docstring mentioning that component isn't supported yet on conda.
* I left one question about pmdarima version

File: evalml/tests/conftest.py
Comment (ParthivNaresh) [reply evalml/tests/conftest.py:115]: @dsherry Yes we'll have a separate PR for the feedstock that adds `--is-using-conda` to the command

File: requirements.txt
Comment (ParthivNaresh) [reply requirements.txt:14]: This unfortunately causes `latest_dependency_versions` to [throw an error](https://github.com/alteryx/evalml/runs/2529399400?check_suite_focus=true) as it expects `1.8.2`, but that version pins `numpy` to a lower version than what we can accept.

File: evalml/pipelines/components/transformers/scalers/standard_scaler.py
Comment (freddyaboulton) [reply evalml/pipelines/components/transformers/scalers/standard_scaler.py:36]: How did this come up? Is this a bug that we've always had or triggered by another change in this PR?

**Pull request merged**
